### PR TITLE
Update electron version

### PR DESCRIPTION
**Pull request by salman90**
## Purpose

*  Updated electron version to the latest release and some additional changes like:

1.  Changed the authentication behaver from redirect to popup.
2.  Added silent authentication and token fetching.
3.  Added contextBridge for more secure interaction.

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-desktop.git
cd App
git checkout update-electron-version
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,15 +28,18 @@ This sample demonstrates the following **MSAL Node** concepts:
...
+| `AppCreationScripts/`        | Contains Powershell scripts for automating app registration. |
+| `App/authProvider.js`        | Main authentication logic resides here.                      |
+| `App/main.js`                | Application main process.                                    |
+| `App/fetch.js`               | Axios HTTP client for calling endpoints with a bearer token. |
+| `App/renderer.js`            | Renderer processes and UI methods.                           |
+| `App/constants.js`           | Example user accounts in JSON .                              |
+| `.CustomProtocolListener.js` | Register and unregister a custom typed protocol.             |
+| `.UIManager.js`              | Environment variables of authentication parameters.          |
+| `.preload.js`                | Give the Renderer process controlled access to some Node API.|
+| `.env`                       | Environment variables of authentication parameters.          |
```
Comment (derisen) [inline README.md]: Redirect URI needs to change from `msal://redirect` to `msal{appId}://auth` (which is also default on Portal desktop app registration) here in the README and elsewhere (.env etc.)

**Review by derisen — APPROVED** [review]
LGTM, one comment @salman90 

Note the merging needs to be coordinated with the tutorial article PR.

File: App/CustomProtocolListener.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  /**
+   * Registers a custom string protocol on which the library will
+   * listen for Auth Code response.
+   */
+  start() {
+    const codePromise = new Promise((resolve, reject) => {
+      protocol.registerStringProtocol(this.host, (req, callback) => {
+        const requestUrl = new URL(req.url);
+        const authCode = requestUrl.searchParams.get("code");
+        authCode
```
Comment (derisen) [inline App/CustomProtocolListener.js]: @salman90 we should unregister the protocol listener if auth code promise does not resolve. see this [PR ](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/4840)

Comment (salman90) [reply App/CustomProtocolListener.js]: Updated, thanks for the feedback!

File: README.md (hunk 1 above)
Comment (salman90) [reply README.md]: This will break the quickstart sample. Let's figure out a solution and fix the issue in a separate branch.

Comment (salman90) [reply README.md]: Updated!

File: README.md [hunk 3]
```diff
@@ -45,28 +48,51 @@ This sample demonstrates the following **MSAL Node** concepts:
...
 
-1. Register a [desktop app that calls web APIs](https://docs.microsoft.com/azure/active-directory/develop/scenario-desktop-app-registration) in the [Azure Portal](https://portal.azure.com).
-    * Navigate to the **Authentication** blade in the app registration.
-    * Select **Add a platform** and choose **Mobile and Desktop application**
-    * For the **redirect URI**, enter `msal://redirect` into the input box.
-1. Clone this repository `git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-desktop.git`
 1. Open the [.env](.env) file and provide the required configuration values.
     1. Replace the string `Enter_the_Application_Id_Here` with your app/client ID on Azure AD portal.
     1. Replace the string `Enter_the_Tenant_Info_Here` with your tenant ID on Azure AD portal.
+    1. Replace the string `Enter_the_Redirect_Uri_Here` with the default redirect uri generated by portal `msal{YourAppId}://auth`
```
Comment (derisen) [inline README.md]:
```suggestion
    1. Replace the string `Enter_the_Redirect_Uri_Here` with the default redirect URI generated by Azure portal app registration e.g. `msal{Your_Application/Client_Id}://auth`
```

File: README.md [hunk 4]
```diff
@@ -45,28 +48,51 @@ This sample demonstrates the following **MSAL Node** concepts:
...
+Clone this repository `git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-desktop.git`
+
+#### Step 3: Configure the Electron sample project
 
-1. Register a [desktop app that calls web APIs](https://docs.microsoft.com/azure/active-directory/develop/scenario-desktop-app-registration) in the [Azure Portal](https://portal.azure.com).
-    * Navigate to the **Authentication** blade in the app registration.
-    * Select **Add a platform** and choose **Mobile and Desktop application**
-    * For the **redirect URI**, enter `msal://redirect` into the input box.
-1. Clone this repository `git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-desktop.git`
 1. Open the [.env](.env) file and provide the required configuration values.
```
Comment (derisen) [inline README.md]:
```suggestion
1. Open the [.env](.env) file and provide the required configuration values.
```

@salman90 Can you make sure you're checking in the .env file?

File: App/CustomProtocolListener.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+          reject(new Error("No code found in URL"));
+        }
+      });
+    });
+
+    return codePromise;
+  }
+
+  /**
+   * Unregisters a custom file protocol to stop listening for Auth Code response.
```
Comment (derisen) [inline App/CustomProtocolListener.js]: custom **string** protocol, here and elsewhere

File: README.md (hunk 4 above)
Comment (derisen) [reply README.md]: my bad, it's checked in

File: README.md (hunk 3 above)
Comment (salman90) [reply README.md]: Updated

File: App/CustomProtocolListener.js (hunk 5 above)
Comment (salman90) [reply App/CustomProtocolListener.js]: done

File: README.md (hunk 4 above)
Comment (salman90) [reply README.md]: replaced .env file with authConfig.js

**Pull request merged**